### PR TITLE
Remove the dependency on Metaflow #82

### PR DESCRIPTION
**Pull request by tdruez**

### CODE REVIEW

File: scanpipe/management/commands/__init__.py [hunk 1]
```diff
@@ -64,6 +64,13 @@ def get_run_status_code(self, run):
         return status
 
 
+def indent(text, by=0, sep="\n"):
+    """
+    Indent each line of the provided `text`.
+    """
+    return sep.join(" " * by + line for line in text.split(sep))
```
Comment (pombredanne) [inline scanpipe/management/commands/__init__.py:71]: Could this be a job for the stdlib textwrap? https://docs.python.org/3/library/textwrap.html

File: scanpipe/management/commands/graph.py [hunk 2]
```diff
@@ -26,19 +26,46 @@
...
 
 def is_graphviz_installed():
-    exitcode, _ = subprocess.getstatusoutput("which dot")
+    exitcode = subprocess.getstatusoutput("which dot")[0]
     if exitcode == 0:
         return True
     return False
 
 
+def pipeline_graph_dot(pipeline_class, fontname="Helvetica", shape="record"):
```
Comment (pombredanne) [inline scanpipe/management/commands/graph.py:42]: I wonder if in the future we could eschew using `dot` at all.

File: scanpipe/management/commands/run.py [hunk 3]
```diff
@@ -30,33 +30,18 @@
...
+        run = self.project.get_next_run()
 
         if not run:
-            raise CommandError(f"No pipelines to {action} on project {self.project}")
-
-        self.stdout.write(f"Pipeline {run.pipeline} {action} in progress...")
-        getattr(run, task_function)()
+            raise CommandError(f"No pipelines to run on project {self.project}")
 
+        self.stdout.write(f"Pipeline {run.pipeline} run in progress...")
```
Comment (pombredanne) [inline scanpipe/management/commands/run.py:41]: What is `self.stdout`?

File: scanpipe/pipelines/docker.py [hunk 4]
```diff
@@ -67,92 +47,77 @@ def extract_layers(self):
...
-        self.next(self.end)
-
-    @step
-    def end(self):
-        """
-        Analysis completed.
-        """
-
 
-if __name__ == "__main__":
```
Comment (pombredanne) [inline scanpipe/pipelines/docker.py:157]: we are missing out the direct execution of a pipeline, but I will not miss this at all. A proper managed CLI is better.

File: scanpipe/tests/pipelines/do_nothing.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+#
+# ScanCode.io is a free software code scanning tool from nexB Inc. and others.
+# Visit https://github.com/nexB/scancode.io for support and download.
+
+from scanpipe.pipelines import Pipeline
+
+
+class DoNothing(Pipeline):
+    """
+    A pipeline that does nothing, in 2 steps.
```
Comment (pombredanne) [inline scanpipe/tests/pipelines/do_nothing.py:28]: :+1:

**Review by pombredanne — APPROVED** [review]
LGTM!

File: scanpipe/management/commands/run.py (hunk 3 above)
Comment (tdruez) [reply scanpipe/management/commands/run.py:41]: This is the default Django management command API https://docs.djangoproject.com/en/3.1/howto/custom-management-commands/#module-django.core.management

File: scanpipe/management/commands/graph.py (hunk 2 above)
Comment (tdruez) [reply scanpipe/management/commands/graph.py:42]: We can get rid of the feature now we have a way to visualize Pipeline in the UI.

File: scanpipe/management/commands/__init__.py (hunk 1 above)
Comment (tdruez) [reply scanpipe/management/commands/__init__.py:71]: Yes, thanks for the tip.

**Comment by pombredanne** [comment]
BTW, we may need to update the documentation accordingly?

**Comment by tdruez** [comment]
@pombredanne the only reference to Metaflow in the docs is this:
> ScanPipe was originally developed using Django, ScanCode Toolkit, and Metaflow to help boost productivity of code analysts who work on a wide variety of software composition analysis projects.

**Pull request merged**